configuration/azure/appconfig/appconfig.go

```diff
@@ -0,0 +1,195 @@
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig"
+	"github.com/google/uuid"
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/dapr/components-contrib/configuration"
+	azauth "github.com/dapr/components-contrib/internal/authentication/azure"
+
+	"github.com/dapr/kit/logger"
+)
+
+const (
+	host                 = "appConfigHost"
+	connectionString     = "appConfigConnectionString"
+	maxRetries           = "maxRetries"
+	retryDelay           = "retryDelay"
+	maxRetryDelay        = "maxRetryDelay"
+	defaultMaxRetries    = 3
+	defaultRetryDelay    = time.Second * 4
+	defaultMaxRetryDelay = time.Second * 120
+)
+
+// ConfigurationStore is a Azure App Configuration store.
+type ConfigurationStore struct {
+	client   *azappconfig.Client
+	json     jsoniter.API
+	metadata metadata
+
+	logger logger.Logger
+}
+
+// NewAzureAppConfigurationStore returns a new Azure App Configuration store.
+func NewAzureAppConfigurationStore(logger logger.Logger) configuration.Store {
+	s := &ConfigurationStore{
+		json:   jsoniter.ConfigFastest,
+		logger: logger,
+	}
+
+	return s
+}
+
+// Init does metadata and connection parsing.
+func (r *ConfigurationStore) Init(metadata configuration.Metadata) error {
+	m, err := parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	r.metadata = m
+
+	coreClientOpts := azcore.ClientOptions{
+		Telemetry: policy.TelemetryOptions{
+			ApplicationID: "dapr-" + logger.DaprVersion,
+		},
+	}
+
+	options := azappconfig.ClientOptions{
+		ClientOptions: coreClientOpts,
+	}
+
+	if r.metadata.connectionString != "" {
+		r.client, err = azappconfig.NewClientFromConnectionString(r.metadata.connectionString, &options)
+		if err != nil {
+			return err
+		}
+	} else {
+		var settings azauth.EnvironmentSettings
+		settings, err = azauth.NewEnvironmentSettings("appconfig", metadata.Properties)
+		if err != nil {
+			return err
+		}
+
+		var cred azcore.TokenCredential
+		cred, err = settings.GetTokenCredential()
+		if err != nil {
+			return err
+		}
+
+		r.client, err = azappconfig.NewClient(r.metadata.host, cred, &options)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func parseMetadata(meta configuration.Metadata) (metadata, error) {
+	m := metadata{}
+
+	if val, ok := meta.Properties[host]; ok && val != "" {
+		m.host = val
+	}
+
+	if val, ok := meta.Properties[connectionString]; ok && val != "" {
+		m.connectionString = val
+	}
+
+	if m.connectionString != "" && m.host != "" {
+		return m, fmt.Errorf("azure store error: can't set both %s and %s fields in metadata", host, connectionString)
+	}
+
+	if m.connectionString == "" && m.host == "" {
+		return m, fmt.Errorf("azure store error: specify %s or %s field in metadata", host, connectionString)
+	}
+
+	m.maxRetries = defaultMaxRetries
```

Comment:
These properties (maxRetries, maxRetryDelay, retryDelay) aren't used and can be removed (also the Azure SDKs handle retrying built-in)

Reply:
SDKs has the defaults values and I think we can give user the options to change them. 

I had missed setting the retry properties in code but this is how options look in the latest commit 👇

``` golang
        coreClientOpts := azcore.ClientOptions{
		Telemetry: policy.TelemetryOptions{
			ApplicationID: "dapr-" + logger.DaprVersion,
		},
		Retry: policy.RetryOptions{
			MaxRetries:    int32(m.maxRetries),
			RetryDelay:    m.maxRetryDelay,
			MaxRetryDelay: m.maxRetryDelay,
		},
	}
```

Reply:
and similarly for other errors ...